src/lib/parts.ts

```diff
@@ -240,7 +251,7 @@ export class NodePart implements Part {
 
   private _commitTemplateResult(value: TemplateResult): void {
     const template = this.options.templateFactory(value);
-    if (this.value && this.value.template === template) {
+    if (this.value instanceof TemplateInstance) {
```

Comment:
This shouldn't be safe to do. Why'd you remove the other check? Is a test failing? If not we should add one.